hivemq-edge/src/main/java/com/hivemq/bootstrap/netty/initializer/AbstractChannelInitializer.java

```diff
@@ -106,6 +108,12 @@ protected void initChannel(final @NotNull T ch) throws Exception {
         //MQTT_5_FLOW_CONTROL_HANDLER is added here after CONNECT
         ch.pipeline()
                 .addLast(MQTT_MESSAGE_BARRIER, new MessageBarrier(channelDependencies.getMqttServerDisconnector()));
+
+        final @Nullable HandlerPackage handlerPackage = channelDependencies.getHandlerProvider().get();
```

Comment:
Not necessary for this, but it could make sense to also offer an insertAfter with the previous handler as param instead of doing addLast, then the position in the pipeline is not fixed to the end